adminSDK/alertcenter/quickstart/src/main/java/AdminSDKAlertCenterQuickstart.java

```diff
@@ -0,0 +1,106 @@
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.alertcenter.v1beta1.AlertCenter;
+import com.google.api.services.alertcenter.v1beta1.model.Alert;
+import com.google.api.services.alertcenter.v1beta1.model.AlertFeedback;
+import com.google.api.services.alertcenter.v1beta1.model.ListAlertsResponse;
+import com.google.auth.Credentials;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+
+public class AdminSDKAlertCenterQuickstart {
+
+  private static final String APPLICATION_NAME = "Google Admin SDK Alert Center API Java Quickstart";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  /**
+   * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+   * your previously saved tokens/ folder.
+   */
+  private static final List<String> SCOPES = Collections
+      .singletonList("https://www.googleapis.com/auth/apps.alerts");
+  private static final String CREDENTIALS_FILE_PATH = "/credentials.json";
+
+
+  /**
+   * Creates an authorized Credentials object.
+   *
+   * @return An authorized Credentials object.
+   * @throws IOException If the credentials.json file cannot be found.
+   */
+  private static Credentials getCredentials(String delegatedAdmin) throws IOException {
+    // [START admin_sdk_alertcenter_get_credentials]
+    InputStream in = AdminSDKAlertCenterQuickstart.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
+    if (in == null) {
+      throw new IOException("Credential file was not found");
+    }
+    GoogleCredentials credentials = ServiceAccountCredentials
+        .fromStream(in)
+        .createDelegated(delegatedAdmin)
+        .createScoped(SCOPES);
+    // [END admin_sdk_alertcenter_get_credentials]
+    return credentials;
+  }
+
+  public static void main(String... args) throws IOException, GeneralSecurityException {
+    // [START admin_sdk_alertcenter_create_client]
+    String delegatedAdmin = "admin@xxx.com";
+    NetHttpTransport transport = GoogleNetHttpTransport.newTrustedTransport();
+    AlertCenter service = new AlertCenter.Builder(transport, JSON_FACTORY,
+        new HttpCredentialsAdapter(getCredentials(delegatedAdmin)))
+        .setApplicationName(APPLICATION_NAME)
+        .build();
+    // [END admin_sdk_alertcenter_create_client]
+
+    // [START admin_sdk_alertcenter_list_alerts]
+    ListAlertsResponse listResponse = service.alerts().list().setPageSize(20).execute();
+    while (listResponse.getAlerts() != null) {
```

Comment:
We tend to use a do-while loop instead, with the pattern:

```
pageToken = null
do:
  response = makeRequest.setPageToken(pageToken);
  ...
  pageToken = response.nextPageToken
while pageToken != null
```

Reply:
Additionally, you should end the loop when the page token is null, not when the results are null. For some APIs you can get a page with no responses in it, but there are still additional pages of results.